storage_delegation/main.tf

```diff
@@ -6,16 +6,50 @@ locals {
     "eu-de"    = "eu-de.dataplatform.cloud.ibm.com",
     "jp-tok"   = "jp-tok.dataplatform.cloud.ibm.com"
   }
-  dataplatform_ui = local.dataplatform_ui_mapping[local.location]
+  dataplatform_ui             = local.dataplatform_ui_mapping[local.location]
+  parsed_kms_key_crn          = var.cos_kms_key_crn != null ? split(":", var.cos_kms_key_crn) : []
+  kms_service                 = length(local.parsed_kms_key_crn) > 0 ? local.parsed_kms_key_crn[4] : null
+  kms_scope                   = length(local.parsed_kms_key_crn) > 0 ? local.parsed_kms_key_crn[6] : null
+  kms_account_id              = length(local.parsed_kms_key_crn) > 0 ? split("/", local.kms_scope)[1] : null
+  kms_key_id                  = length(local.parsed_kms_key_crn) > 0 ? local.parsed_kms_key_crn[9] : null
+  target_resource_instance_id = split(":", var.cos_kms_crn)[7]
```

Comment:
Please use crn parser - https://github.com/terraform-ibm-modules/terraform-ibm-common-utilities/tree/main/modules/crn-parser

Reply:
This has been implemented